src/app/components/delegate-registration/delegateRegistration.js

```diff
@@ -0,0 +1,58 @@
+import './delegateRegistration.less';
+
+app.directive('delegateRegistration', ($mdDialog, $peers, Account, success) => {
+  const DelegateRegistrationLink = function ($scope, $element) {
+    $scope.form = {
+      name: '',
+      fee: 25,
+      error: '',
+      onSubmit: (form) => {
+        if (form.$valid) {
+          $peers.active.registerDelegate($scope.form.name.toLowerCase(), Account.get().passphrase)
+            .then(() => {
+              $scope.reset(form);
+
+              success.dialog({ text: 'Account was successfully registered as delegate.' })
```

Comment:
This message should rather say that the registration is pending and will be processed with the next block.